### PR TITLE
IWF-107: Updating iwf-idl submodule to reflect changes for IdReusePol…

### LINKED ISSUE #204: Fix typo IdReusePolicy of ALLOW_IF_PREVIOUS_EXISTS_ABNORMALLY
1. to fix the typo https://github.com/indeedeng/iwf-idl/issues/45 with backward compatibility
2. provide more comment/java docs for this policy

### PR DESCRIPTION
**Pull request by stevo89519**
…icy typo changes

### Checklist
- [ ] Code compiles correctly
- [ ] Tests for the changes have been added
- [ ] All tests passing

### Related Issue
Closes #204

### CODE REVIEW

**Comment by stevo89519** [comment]
@longquanzheng I wasn't able to find any references to ALLOW_IF_PREVIOUS_EXISTS_ABNORMALLY in this repo except for within the iwf-idl submodule, so I just updated it to reflect the changes made previously.
Did I miss something?

**Review by longquanzheng — APPROVED** [review]

**Comment by longquanzheng** [comment]
> @longquanzheng I wasn't able to find any references to ALLOW_IF_PREVIOUS_EXISTS_ABNORMALLY in this repo except for within the iwf-idl submodule, so I just updated it to reflect the changes made previously. Did I miss something?

I think that’s right. Because we don’t have any tests or code directly referencing to it. We allow client to pass the policy in workflow options. 
But would encourage add a new test ☺️

**Comment by stevo89519** [comment]
> But would encourage add a new test so that you will understand it better ☺️

@longquanzheng Test added. Please re-review.

File: src/test/java/io/iworkflow/integ/basic/AbnormalExitState1.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package io.iworkflow.integ.basic;
+
+import io.iworkflow.core.Context;
+import io.iworkflow.core.StateDecision;
+import io.iworkflow.core.WorkflowState;
+import io.iworkflow.core.command.CommandResults;
+import io.iworkflow.core.communication.Communication;
+import io.iworkflow.core.persistence.Persistence;
+
+public class AbnormalExitState1 implements WorkflowState<Integer> {
```
[the remark this answers is not in the document]
Comment (stevo89519) [reply src/test/java/io/iworkflow/integ/basic/AbnormalExitState1.java:12]: Thanks! I added it.

**Pull request merged**